codice/alliance
# [1.1.x] CAL-459 Updated how attributes are populated with from the ImageSegment.TGTID

#### What does this PR do?
The `ImageSegment.TGTID` may contain a country code that should be used to populate `location.country-code`. This is not easily testable because I would think that if a NITF image segment had a location, then the GeoCoderPlugin would use the `location` attribute to populate the 
`location.country-code`.

Additionally, while the `isr.target-id` attribute contains the stripped value from `ImageSegment.TGTID`, this PR updates the `ext.nitf.image.target-identifier` to contain the original, non-stripped value. This is important because the `ImageSegment.TGTID` consists of 3 parts (characters 0-9, 10-14, 15-16) where each part is optionally stored in the string value as spaces. After stripping the `ImageSegment.TGTID`, it is not obvious which of the 3 parts is left, so it would be worthwhile to keep the original value in at least the non-normalized attribute.
#### Who is reviewing it? 
@peterhuffer @kcover @ethantmanns @JGatley 
#### Choose 2 committers to review/merge the PR.
@clockard
@vinamartin
#### How should this be tested?

- Ingest a NITF with a populated country code in the `ImageSegment.TGTID`. Confirm that the `location.country-code` is populated appropriately.
- Ingest a NITF with a partially-populated `ImageSegment.TGTID`. Confirm that the `ext.nitf.image.target-identifier` contains the original, non-stripped value. @kcover noticed that repeated spaces appear as one space in Intrigue or when `Export as XML`. You can confirm the number of spaces is correct by downloading the metacard xml, instead of just viewing it in the browser.

#### What are the relevant tickets?
[CAL-459](https://codice.atlassian.net/browse/CAL-459)
#### Checklist:
